# Exercício 18 (MINI BOSS): Análise de Desempenho Escolar

#### Você é um cientista de dados contratado por uma escola para analisar o desempenho de três turmas (Turma A, Turma B, Turma C) em duas disci plinas: Matemática e Ciências.
#### O objetivo é gerar um relatório completo utilizando as medidas estatísticas estudadas para entender o desempenho geral dos alunos e descobrir padrões importantes.
#### Tarefas:

#### 1. Carregar o dataset que contém as notas dos alunos das três turmas. O arquivo chama 18_analise_avancada.csv.

In [2]:
import pandas as pd

students_notes = pd.read_csv("../18_analise_avancada.csv")
students_notes

Turma  Matemática  Ciências
0      C          83        96
1      A          47        91
2      C          63        52
3      C          50        80
4      A          90        42
..   ...         ...       ...
95     A          78        79
96     A          72        78
97     C          40        57
98     A          96        79
99     A          66        40

[100 rows x 3 columns]

#### 2. Calcular as seguintes medidas estatísticas para cada turma (A, B e C) em Matemática e Ciências:
#### a. Média

In [9]:
df_media = students_notes.groupby(['Turma']).mean()

print("A média em matemática e ciências das turmas A, B e C são, respectivamente:")
df_media

A média em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática   Ciências
Turma                       
A       69.515152  70.393939
B       70.888889  71.555556
C       68.741935  67.580645

#### b. Mediana

In [10]:
df_median = students_notes.groupby(['Turma']).median()
print("A mediana em matemática e ciências das turmas A, B e C são, respectivamente:")
df_median

A mediana em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática  Ciências
Turma                      
A            68.0      72.0
B            71.0      72.0
C            74.0      67.0

#### c. Moda

In [16]:
df_mode = students_notes.groupby('Turma').apply(lambda x: x.mode(), include_groups=False)
df_mode = df_mode.dropna()

print("A moda em matemática e ciências das turmas A, B e C são, respectivamente:")
df_mode

A moda em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática  Ciências
Turma                        
A     0        94.0      71.0
B     0        54.0      58.0
C     0        78.0      42.0
      1        83.0      45.0

#### d. Amplitude (diferença entre a maior e a menor nota)

In [75]:
df_max = students_notes.groupby('Turma').max()
df_min = students_notes.groupby('Turma').min()

df_amplitude = df_max - df_min

print("A amplitude em matemática e ciências das turmas A, B e C são, respectivamente:")
df_amplitude

A amplitude em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática  Ciências
Turma                      
A              57        58
B              53        58
C              60        57

#### e. Variância

In [8]:
df_variance = students_notes.groupby('Turma').var()

print("A variância em matemática e ciências das turmas A, B e C são, respectivamente:")
df_variance

A variância em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática    Ciências
Turma                        
A      325.570076  345.996212
B      234.101587  374.311111
C      307.997849  347.318280

#### f. Desvio Padrão

In [78]:
df_std = students_notes.groupby('Turma').std()

print("O desvio padrão em matemática e ciências das turmas A, B e C são, respectivamente:")
df_std

O desvio padrão em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática   Ciências
Turma                       
A       18.043561  18.600973
B       15.300379  19.347122
C       17.549868  18.636477

#### g. Covariância e Correlação entre Matemática e Ciências

In [10]:
df_covariance = students_notes.groupby('Turma').cov()

print("A covariância em matemática e ciências das turmas A, B e C são, respectivamente:")
df_covariance

A covariância em matemática e ciências das turmas A,} B e C são, respectivamente:


Matemática    Ciências
Turma                                   
A     Matemática  325.570076  -68.646780
      Ciências    -68.646780  345.996212
B     Matemática  234.101587  -25.165079
      Ciências    -25.165079  374.311111
C     Matemática  307.997849   77.054839
      Ciências     77.054839  347.318280

In [32]:
df_correlation = students_notes.groupby('Turma').corr()

print("A correlação em matemática e ciências das turmas A, B e C são, respectivamente:")
df_correlation

A correlação em matemática e ciências das turmas A, B e C são, respectivamente:


Matemática  Ciências
Turma                                 
A     Matemática    1.000000 -0.204532
      Ciências     -0.204532  1.000000
B     Matemática    1.000000 -0.085012
      Ciências     -0.085012  1.000000
C     Matemática    1.000000  0.235593
      Ciências      0.235593  1.000000

# Conclusões:
### 1. Com base nas medidas acima, qual turma teve o melhor desempenho geral em Matemática e Ciências?

#### Em matemática, as Turmas que mais se destacam são a Turma A e a Turma C aparecendo 4 vezes com maiores valores das medidas
#### Em Ciências, a Turma que mais se destacou foi a Turma B aparecendo 5 vezes com maiores valores das medidas

### 2. Qual turma tem as notas mais consistentes em termos de desvio padrão?

#### A turma B, possuindo um desvio de padrão de 19.3

### 3. Existe uma correlação entre o desempenho em Matemática e Ciências para cada turma? Essa correlação é positiva ou negativa?

#### Existe sim, sendo uma correlação positiva mas não perfeita

### 4. Qual turma apresenta a maior amplitude de notas? O que isso nos diz sobre a desigualdade de desempenho dentro da turma?

#### A turma C possui a maior amplitude de todas, sendo 60 em Matemática. Mas as Turmas A e B tem a maior amplitude em ciências, 58 respectivamente para as duas, apenas um número maior que a da Turma C (57). Isso mostra que a Turma C teve resultados bem diferentes em valores em matemática, mas manteve um certo padrão em Ciências, quase que igualmente às outras turmas.